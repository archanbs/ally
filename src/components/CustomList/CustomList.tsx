import React from "react";
import { AiOutlineClose } from "react-icons/ai";

export type CustomListProps = {
  label: string;
  items: string[];
  onItemClicked?: (itemName: string, index: number) => void;
  showClearAll?: boolean;
  onClearAllClicked?: () => void;
};

export const CustomList = ({ label, items, onItemClicked, onClearAllClicked, showClearAll }: CustomListProps) => {
  return (
    <div className="flex gap-2 items-center">
      <div>{label}</div>
      <ul className="flex gap-1 flex-wrap">
        {items.map((itemName, index) => {
          return (
            <li
              className="p-1 px-2 text-[12px] my-2 rounded-2xl bg-themeBlue text-center text-white flex"
              style={{ cursor: onItemClicked ? "pointer" : "auto" }}
              key={`filter-${index}`}
              onClick={(ev) => {
                console.log("remove item", itemName);
                ev.preventDefault();
                onItemClicked?.(itemName, index);
              }}
              tabIndex={1}
            >
              {itemName}
              <AiOutlineClose className="self-center ml-2" />
            </li>
          );
        })}
      </ul>
      {showClearAll && onClearAllClicked && (
        <div
          data-testId="clearAllButton"
          className="flex text-[#1560bd] cursor-pointer"
          onClick={(ev) => {
            ev.preventDefault();
            onClearAllClicked();
          }}
          tabIndex={1}
        >
          Clear all
        </div>
      )}
    </div>
  );
};
