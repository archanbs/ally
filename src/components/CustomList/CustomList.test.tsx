import renderer from "react-test-renderer";
import { CustomList } from "./CustomList";
import React from "react";
import { render, screen } from "@testing-library/react";

test("CustomList snapshot test", () => {
  const compTree = renderer.create(<CustomList label="Custom list:" items={["item 1", "item 2"]} />).toJSON();
  expect(compTree).toMatchSnapshot();
});

test("CustomList snapshot test with clear all", () => {
  const compTree = renderer
    .create(<CustomList label="Custom list:" items={["item 1", "item 2"]} showClearAll onClearAllClicked={() => console.log("Clear all clicked")} />)
    .toJSON();
  expect(compTree).toMatchSnapshot();
});

test("CustomList does render clear all", () => {
  render(<CustomList label="Custom list:" items={["item 1", "item 2"]} showClearAll onClearAllClicked={() => console.log("Clear all clicked")} />);

  const clearAllButton = screen.getByTestId("clearAllButton");
  expect(clearAllButton).toBeInTheDocument();
});
