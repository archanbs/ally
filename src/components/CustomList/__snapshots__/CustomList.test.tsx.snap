// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomList snapshot test 1`] = `
<div
  className="flex gap-2 items-center"
>
  <div>
    Custom list:
  </div>
  <ul
    className="flex gap-1 flex-wrap"
  >
    <li
      className="p-1 px-2 text-[12px] my-4 rounded-2xl bg-themeBlue text-center text-white flex"
      onClick={[Function]}
      tabIndex={1}
    >
      item 1
      <svg
        className="self-center ml-2"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
        />
      </svg>
    </li>
    <li
      className="p-1 px-2 text-[12px] my-4 rounded-2xl bg-themeBlue text-center text-white flex"
      onClick={[Function]}
      tabIndex={1}
    >
      item 2
      <svg
        className="self-center ml-2"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
        />
      </svg>
    </li>
  </ul>
</div>
`;

exports[`CustomList snapshot test with clear all 1`] = `
<div
  className="flex gap-2 items-center"
>
  <div>
    Custom list:
  </div>
  <ul
    className="flex gap-1 flex-wrap"
  >
    <li
      className="p-1 px-2 text-[12px] my-4 rounded-2xl bg-themeBlue text-center text-white flex"
      onClick={[Function]}
      tabIndex={1}
    >
      item 1
      <svg
        className="self-center ml-2"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
        />
      </svg>
    </li>
    <li
      className="p-1 px-2 text-[12px] my-4 rounded-2xl bg-themeBlue text-center text-white flex"
      onClick={[Function]}
      tabIndex={1}
    >
      item 2
      <svg
        className="self-center ml-2"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
        />
      </svg>
    </li>
  </ul>
  <div
    className="flex text-[#1560bd] cursor-pointer"
    data-testId="clearAllButton"
    onClick={[Function]}
    tabIndex={1}
  >
    Clear all
  </div>
</div>
`;
